bridge-history-api/internal/logic/l1_event_parser.go

```diff
@@ -283,26 +283,19 @@ func (e *L1EventParser) ParseL1MessageQueueEventLogs(logs []types.Log, l1Deposit
 	return l1MessageQueueEvents, nil
 }
 
-func getRealFromAddress(ctx context.Context, eventSender common.Address, client *ethclient.Client, txHash common.Hash, gatewayRouterAddr string) (string, error) {
-	from := eventSender.String()
-	if from == gatewayRouterAddr {
-		tx, isPending, rpcErr := client.TransactionByHash(ctx, txHash)
-		if rpcErr != nil || isPending {
-			log.Error("Failed to get transaction or the transaction is still pending", "rpcErr", rpcErr, "isPending", isPending)
-			return "", rpcErr
-		}
-		// Case 1: deposit/withdraw ETH: EOA -> multisig -> gateway router -> messenger.
-		if tx.To() != nil && (*tx.To()).String() != gatewayRouterAddr {
-			return (*tx.To()).String(), nil
-		}
-		// Case 2: deposit/withdraw ETH: EOA -> gateway router -> messenger.
-		signer := types.LatestSignerForChainID(new(big.Int).SetUint64(tx.ChainId().Uint64()))
-		sender, err := signer.Sender(tx)
-		if err != nil {
-			log.Error("Get sender failed", "chain id", tx.ChainId().Uint64(), "tx hash", tx.Hash().String(), "err", err)
-			return "", err
-		}
-		return sender.String(), nil
+func getRealFromAddress(eventSender common.Address, eventMessage []byte, gatewayRouterAddr string) (string, error) {
+	if eventSender != common.HexToAddress(gatewayRouterAddr) {
+		return eventSender.String(), nil
 	}
-	return from, nil
+
+	// deposit/withdraw ETH: EOA -> contract 1 -> ... -> contract n -> gateway router -> messenger.
+	if len(eventMessage) < 32 {
```

Comment:
maybe we should not return error, since sepolia is running a version without such fix.

Reply:
aha, makes sense, I'll change it to a warning log.
currently, the sender could be the gateway router address in sepolia because the contract has been upgraded once.

Reply:
added back the legacy logic as a fallback, cannot handle 100% of cases before the next upgrade, but it's better than nothing.
commit: [9218957](https://github.com/scroll-tech/scroll/pull/1103/commits/9218957717869110924328803d8e6807cdb6630d)

Reply:
cannot handle the case that in current sepolia (if 1 and 2 happen simultaneously):
1. multiple contracts: deposit/withdraw ETH: EOA -> contract 1 -> ... -> contract n -> gateway router.
2. data field >= 32 bytes/

Reply:
yeah, let's hope no one has used the gateway router with data field >= 32 bytes.